Src/MakingSense.SafeBrowsing/GoogleSafeBrowsing/GoogleSafeBrowsingUpdater.cs

```diff
@@ -0,0 +1,171 @@
+        {
+            _configuration = configuration;
+
+            _safebrowsingService = new SafebrowsingService(new BaseClientService.Initializer
+            {
+                ApiKey = _configuration.ApiKey,
+            });
+
+            _database = new GoogleSafeBrowsingDatabase();
+        }
+
+        /// <summary>
+        /// Updates Google Safe Browsing lists based on the remote resource
+        /// </summary>
+        /// <returns></returns>
+        public async Task UpdateAsync()
+        {
+            if (!_database.AllowRequest)
+            {
+                return;
+            }
+
+            var request = new FetchThreatListUpdatesRequest
+            {
+                Client = new ClientInfo
+                {
+                    ClientId = _configuration.ClientId,
+                    ClientVersion = _configuration.ClientVersion
+                },
+                ListUpdateRequests = _database.SuspiciousLists.Select(list => new ListUpdateRequest
+                {
+                    PlatformType = ANY_PLATFORM,
+                    ThreatType = list.Key,
+                    ThreatEntryType = URL,
+                    State = list.Value.State,
+                    Constraints = new Constraints
+                    {
+                        Region = _configuration.Region,
+                        SupportedCompressions = new List<string> { RAW }
+                    }
+                }).ToList()
+            };
+
+            var response = await _safebrowsingService.ThreatListUpdates.Fetch(request).ExecuteAsync();
+
+            _database.MinimumWaitDuration = TimeSpan.FromSeconds(double.Parse((response.MinimumWaitDuration as string).TrimEnd('s')));
+            _database.Updated = DateTimeOffset.Now;
+
+            foreach (var listUpdate in response.ListUpdateResponses)
+            {
+                if(_database.SuspiciousLists[listUpdate.ThreatType].State == listUpdate.NewClientState)
+                {
+                    continue;
+                }
+
+                IEnumerable<byte[]> hashes = _database.SuspiciousLists[listUpdate.ThreatType].Hashes;
+
+                var newHashes = listUpdate.Additions != null ? 
+                                listUpdate.Additions.Select(x => x.RawHashes)
+                                    .SelectMany(rh => SplitByteList(Convert.FromBase64String(rh.RawHashesValue), rh.PrefixSize.Value)) : 
+                                new List<byte[]>();
+
+                if (listUpdate.ResponseType == FULL_UPDATE)
+                {
+                    hashes = newHashes;
+                }
+                else
+                {
+                    if(listUpdate.Removals != null)
+                    {
+                        var indices = listUpdate.Removals.SelectMany(x => x.RawIndices.Indices);
+
+                        hashes = hashes.Where((x, index) => !indices.Contains(index));
```

Comment:
Are you sure that we are making honor to the server indices?

Reply:
Yes, we are keeping the lists lexicographically-sorted to be able to correctly process the removal indices.

Anyway, it is pending to implement the Checksum validation to ensure the whole list is correct.